pandas/core/arrays/categorical.py

```diff
@@ -1234,6 +1234,11 @@ def map(self, mapper):
                                    categories=new_categories,
                                    ordered=self.ordered)
         except ValueError:
+            # NA values are represented in self._codes with -1
+            # np.take causes NA values to take final element in new_categories
+            if any(self._codes == -1):
```

Comment:
Should this use `np.any`? I think `any` will short-circuit, but `np.any` will likely be faster.